src/Adapters/Auth/AuthAdapter.js

```diff
@@ -1,20 +1,102 @@
 /*eslint no-unused-vars: "off"*/
+
+/**
+ * @interface ParseAuthResponse
+ * @property {Boolean} [doNotSave] If true, Parse Server will do not save provided authData.
+ * @property {Object} [response] If set, parse server will send the provided response to the client under authDataResponse
+ * @property {Object} [save] If set, parse server will save the object provided into this key, instead of client provided authData
+ */
+
+/**
+ * AuthPolicy
+ * default: can be combined with ONE additional auth provider if additional configured on user
+ * additional: could be only used with a default policy auth provider
+ * solo: Will ignore ALL additional providers if additional configured on user
+ * @typedef {"default" | "additional" | "solo"} AuthPolicy
+ */
+
 export class AuthAdapter {
-  /*
-  @param appIds: the specified app ids in the configuration
-  @param authData: the client provided authData
-  @param options: additional options
-  @returns a promise that resolves if the applicationId is valid
+  constructor() {
+    /**
+     * Usage policy
+     * @type {AuthPolicy}
+     */
+    this.policy = 'default';
+  }
+  /**
+   * @param appIds The specified app ids in the configuration
+   * @param {Object} authData The client provided authData
+   * @param {Object} options Additional options
+   * @param {Parse.Cloud.TriggerRequest} request
+   * @param {Object} config
+   * @returns {(Promise<undefined|void>|void|undefined)} resolves or returns if the applicationId is valid
+   */
+  validateAppId(appIds, authData, options, request) {
+    return Promise.resolve({});
+  }
+
+  /**
+   * Legacy usage, if provided it will be triggered when authData related to this provider is touched (signup/update/login)
+   * otherwise you should implement validateSetup, validateLogin and validateUpdate
+   * @param {Object} authData The client provided authData
+   * @param {Object} options Additional options
+   * @param {Parse.Cloud.TriggerRequest} request
+   * @param {Object} config
+   * @returns {Promise<ParseAuthResponse|void|undefined>}
+   */
+  validateAuthData(authData, options, request, config) {
+    return Promise.resolve({});
+  }
+
+  /**
+   * Triggered when user provide for the first time this auth provider
+   * could be a register or the user adding a new auth service
+   * @param {Object} authData The client provided authData
+   * @param {Object} options Additional options
+   * @param {Parse.Cloud.TriggerRequest} request
+   * @param {Object} config
+   * @returns {Promise<ParseAuthResponse|void|undefined>}
+   */
+  validateSetUp(authData, options, req, user) {
+    return Promise.resolve({});
+  }
+
+  /**
+   * Triggered when user provide authData related to this provider
+   * he is not logged in and has already set this provider before
```

Comment:
Rephrase... "he"?

- end lines with dot.
- use backticks for code (vars, methods)
- missing param descriptions